torch/csrc/utils/device_lazy_init.cpp

```diff
@@ -77,13 +78,18 @@ void set_device_in_bad_fork(at::DeviceType device_type, bool value) {
 // Should be called before the first device runtime call.
 void register_fork_handler_for_device_init(at::DeviceType device_type) {
 #ifndef WIN32
-  auto& flag = at_fork_once_flags[static_cast<int>(device_type)];
-  c10::call_once(flag, [device_type]() {
-    static at::DeviceType at_fork_device_type = device_type;
+  at_fork_registered[static_cast<int>(device_type)] = true;
+  c10::call_once(at_fork_register_once, []() {
     pthread_atfork(nullptr, nullptr, []() {
-      set_device_in_bad_fork(at_fork_device_type, true);
-      if (is_device_lazy_init_supported(at_fork_device_type)) {
-        set_requires_device_init(at_fork_device_type, true);
+      for (int i = 0; i < static_cast<int>(at::COMPILE_TIME_MAX_DEVICE_TYPES); ++i) {
```

Comment:
Moreover, I don't know how to implement it more clearly or in a more understandable way, although I don`t think using a loop is a good way.